src/eth1spec/evm/instructions.py

```diff
@@ -0,0 +1,91 @@
+"""
+Ethereum Virtual Machine (EVM) Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the instructions understood by the EVM.
+"""
+
+
+from ..base_types import U256
+from . import Evm
+from .gas import GAS_VERY_LOW, subtract_gas
+from .stack import pop, push
+
+
+def add(evm: Evm) -> None:
+    """
+    Adds the top two elements of the stack together, and pushes the result back
+    on the stack.
+
+    Parameters
+    ----------
+    evm : `Evm`
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+
+    x = pop(evm.stack)
+    y = pop(evm.stack)
+
+    val = x + y
```

Comment:
I'm assuming it is intentional that this leaves out the overflow wrap-around.

Reply:
Nope, this should've used `wrapping_add`. I didn't realize we were throwing a `ValueError` on overflow.